### ISSUE TITLE
SnappyData RowStore Quickstart not working

### ISSUE THREAD

**Issue by rolandjohann** [SnappyDataInc/snappydata#555]
Following the steps from http://rowstore.docs.snappydata.io/docs/getting_started/15-minutes.html leads to following excetpion when trying to connect to snappy cluster via `snappy` binary:

```
SnappyData Version 0.8
snappy> connect client '127.0.0.1:1527';
Apr 07, 2017 2:13:34 PM java.util.logging.LogManager$RootLogger log
INFORMATION: Starting client on '192.168.1.193' with ID='71533|2017/04/07 14:13:34.049 MESZ' Source-Revision=392bbf24c28530a624b81e1f95232b6da6688e53
FEHLER XJ001: (SQLState=XJ001 Severity=0) (127.0.0.1[1527]) Java-Ausnahme: 'ASSERT FAILED Number of parameters expected for message id 40XD0 (0) does not match number of arguments received (2): class com.pivotal.gemfirexd.internal.shared.common.sanity.AssertFailure'.
Caused by: AssertFailure: ASSERT FAILED Number of parameters expected for message id 40XD0 (0) does not match number of arguments received (2)
	at com.pivotal.gemfirexd.internal.shared.common.sanity.SanityManager.ASSERT(SanityManager.java:260)
Usage:

  CONNECT CLIENT '<host>:<port>[;<properties>]' [ AS connectionName ];
                               -- connects to a server at given host/port,
                               -- with optional semi-colon separated properties

    Example: CONNECT CLIENT 'localhost:1527';
             -- connects to server running on localhost at port 1527

    Example: CONNECT CLIENT 'localhost:1527;user=gem1;password=gem1';
             -- connects to server running on localhost:1527 with user/password
```

when starting snappy data with `snappy-start-all.sh` everything works fine, but `snappy-start-all.sh rowstore` won't work.

What I actually try to do is HDFS read/write for tables. When executing a `CREATE HDFSSTORE [...]` query an exception occurs hinting that HDFSSTORE is not supported when running snappy data without rowstore.

Maybe it's similar to #531.

**Comment by dshirish** [comment]
SnappyData tables currently do not support HDFS read/write, though Spark APIs can be used to read HDFS data and load into SnappyData.

The pure rowstore mode is mainly intended for the users of GemFire XD transitioning to SnappyData.  Also the rowstore quickstart works when the the cluster is started in rowstore mode, that is with-

```
snappy-start-all.sh rowstore
```
Also on latest released version (0.8) the snappy shell should be launched using the rowstore option when cluster is launched in rowstore mode. So the steps in rowstore quickstart will need to use following whenever snappy shell is launched. 

```
bin/snappy-sql rowstore
```

This is because the latest 0.8 release by default uses new Thrift based JDBC driver as opposed to earlier DRDA JDBC driver. Starting the the snappy-sql shell with rowstore option ensures that the DRDA driver is used for the rowstore mode.

**Comment by rmmjohann** [comment]
@dshirish Thanks for quick reply!

With `bin/snappy-sql rowstore` it works.

Our intent is to use SnappyData as In-Memory database and data lake with upsert abilities as well.
For this we want to use the rowstore HDFS read/write with proper eviction.

If we understand correctly SnappyData uses either GemfireXD or RowStore, where RowStore is the default implementation. Is that correct?

We maybe also will use spark for ingestion into snappy tables (backed by HDFS) so at our case it is not relevant that we can read from HDFS outside of snappy data.

**Comment by piercelamb** [comment]
@rmmjohann the in-memory database inside SnappyData is an extension of GemFireXD. Rowstore was created for existing customers that were using GemFireXD to enable them to transition to SnappyData more easily. Using SnappyData in RowStore mode is sort of like saying "I want to use SnappyData like an operational SQL database, but do not really care about using Spark." We intended it for a niche community, but have found that people often want to try Rowstore because much of the GemFireXD documentation can be used along with it.

If you are planning to use the Spark features in SnappyData, we would advise to use SnappyData in the normal, non-rowstore mode.

**Comment by sumwale** [comment]
@rmmjohann Update/delete support for column tables will be in next release and is being tracked in JIRA: https://jira.snappydata.io/browse/SNAP-1387 . The need for in-built HDFS persistence has been raised elsewhere too (like issue #531) and is being tracked. Currently you can use Spark's APIs like DataFrameWriter for HDFS persistence in parquet or other formats.

**Comment by sumwale** [comment]
@rolandjohann Support for update/delete of column tables is available in 1.0 RC (final 1.0 release will be available later this week). Direct HDFS persistence (in parquet format) for column tables is being tracked in #400. For now explicit call via Spark Dataset API can be used e.g. session.table("...").write.format("parquet").option("path", "hdfs://...").saveAsTable("..."). This gives a snapshot of data (with partition-wise snapshot isolation semantics for snappydata table reads).
